frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/RelativeDatePicker.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import { Divider, Group, Tabs } from "metabase/ui";
+import { BackButton } from "../BackButton";
+import type { RelativeDatePickerValue } from "../types";
+import { DEFAULT_VALUE, TABS } from "./constants";
+import {
+  getDirection,
+  isIntervalValue,
+  isOffsetIntervalValue,
+  setDirection,
+} from "./utils";
+import { CurrentDatePicker } from "./CurrentDatePicker";
+import { DateIntervalPicker } from "./DateIntervalPicker";
+import { DateOffsetIntervalPicker } from "./DateOffsetIntervalPicker";
+import { TabList } from "./RelativeDatePicker.styled";
+
+interface RelativeDatePickerProps {
+  value: RelativeDatePickerValue | undefined;
+  canUseRelativeOffsets: boolean;
+  onChange: (value: RelativeDatePickerValue) => void;
+  onBack: () => void;
+}
+
+export function RelativeDatePicker({
+  value: initialValue,
+  canUseRelativeOffsets,
+  onChange,
+  onBack,
+}: RelativeDatePickerProps) {
+  const [value, setValue] = useState(initialValue ?? DEFAULT_VALUE);
+  const direction = getDirection(value);
+  const isNew = initialValue == null;
+
+  const handleTabChange = (tabValue: string | null) => {
+    const tab = TABS.find(tab => tab.direction === tabValue);
+    if (tab) {
+      setValue(setDirection(value, tab.direction));
+    }
+  };
```

Comment:
Is `tabValue` expected to be `null`?

Reply:
https://mantine.dev/core/tabs/#tab-deactivation

By default not but the type requires `null`